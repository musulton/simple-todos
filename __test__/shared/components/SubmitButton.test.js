import {render, screen, fireEvent} from "@testing-library/react-native";

import SubmitButton from "../../../src/shared/components/SubmitButton";

describe("SubmitButton", () => {
    test("should render correctly", () => {
        const mockProps = {
            title: "Title",
            onSubmit: jest.fn()
        }

        render(<SubmitButton {...mockProps} />)

        expect(screen.toJSON()).toMatchSnapshot()
    })

    test("should call onSubmit props when button is clicked", () => {
        const mockProps = {
            title: "Title",
            onSubmit: jest.fn()
        }

        render(<SubmitButton {...mockProps} />)
        fireEvent.press(screen.getByText(mockProps.title))

        expect(mockProps.onSubmit).toHaveBeenCalled()
    })
})
