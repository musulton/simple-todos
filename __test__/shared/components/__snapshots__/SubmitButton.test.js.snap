// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitButton should render correctly 1`] = `
<View
  style={
    {
      "alignItems": "flex-end",
    }
  }
>
  <View
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "borderColor": "rgba(0,0,0,.1)",
        "borderWidth": 1,
        "height": 50,
        "justifyContent": "center",
        "marginRight": 20,
        "marginTop": 15,
        "paddingLeft": 20,
        "paddingRight": 20,
        "width": 200,
      }
    }
  >
    <Text
      style={
        {
          "color": "#666666",
          "fontWeight": "600",
        }
      }
    >
      Title
    </Text>
  </View>
</View>
`;
